go-controller/pkg/node/gateway_shared_intf.go

```diff
@@ -853,7 +865,7 @@ func (npw *nodePortWatcher) AddService(service *corev1.Service) error {
 	if exists := npw.addOrSetServiceInfo(name, service, hasLocalHostNetworkEp, localEndpoints); !exists {
 		klog.V(5).Infof("Service Add %s event in namespace %s came before endpoint event setting svcConfig",
 			service.Name, service.Namespace)
-		if err := addServiceRules(service, netInfo, localEndpoints, hasLocalHostNetworkEp, npw); err != nil {
+		if err := addServiceRules(service, localEndpoints, hasLocalHostNetworkEp, npw, npw.networkManager); err != nil {
```

Comment:
Wait, so here we already have `netInfo`, but we're passing `npw.networkManager` as input so that `addServiceRules` can obtain `netInfo` once again? Would it be a bad idea to have `addServiceRules` take an optional `netInfo` that we can pass here?